font/src/ft/mod.rs

```diff
@@ -248,21 +248,16 @@ impl FreeTypeRasterizer {
         // Hash pattern together with request pattern to include requested font size in the hash
         let primary_font_key = FontKey::from_pattern_hashes(hash, primary_font.hash());
 
-        // Return font if we already loaded it
-        if self.faces.get(&primary_font_key).is_some() {
-            return Ok(primary_font_key);
-        }
+        // Load font if we haven't loaded one
+        if self.fallback_lists.get(&primary_font_key).is_none() {
```

Comment:
Early return here, please `contains_key` instead of actually accessing the value and then calling `is_some()` or `is_none()`.

Reply:
I did this thing on purpose, because clippy has a bug when using `contains_key` here and then `insert`, like it asks you to use `entry`, but it doesn't make any sense, since you should compute the thing before insert.